graphql_compiler/interpreter/expression_ops/immediate_expression_handlers.py

```diff
@@ -19,28 +20,34 @@ def evaluate_local_field(
     adapter: InterpreterAdapter[DataToken],
     query_metadata_table: QueryMetadataTable,
     query_arguments: Dict[str, Any],
+    current_location: Optional[Location],
     current_type_name: str,
     expression: LocalField,
     data_contexts: Iterable[DataContext],
 ) -> Iterator[Tuple[DataContext, Any]]:
-    # TODO(predrag): Add hints here.
+    # TODO(bojanserafimov): Memoize hints for each location.
+    hints = construct_hints_for_location(query_metadata_table, query_arguments, current_location)
```

Comment:
This is not a legal call, `current_location` is `Optional[Location]` but is being passed to a function that expects `Location`.

`current_location` is only `None` in the global scope, which cannot have `LocalField` expressions, so you can assert that `current_location` is not `None` here.